packages/edit-navigation/src/store/selectors.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { invert } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { createRegistrySelector } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { KIND, POST_TYPE, buildNavigationPostId } from './utils';
+
+export const getNavigationPost = createRegistrySelector(
+	( select ) => ( state, menuId ) => {
+		return select( 'core' ).getEditedEntityRecord(
+			KIND,
+			POST_TYPE,
+			buildNavigationPostId( menuId )
+		);
+	}
+);
+
+export const isResolvingNavigationPost = ( state, menuId ) => {
+	const post = getNavigationPost( state, menuId );
+	return ! post || post.meta.isResolving;
+};
```

Comment:
Storing `isResolving` in meta is not ideal, keeping a proper "resolution" state could potentially solve the rogue request for the fake post.